[Feat] Insurance 도메인 설계

jher235 opened the pull request:
## Related issue 🛠
- closed #14


## Work Description 📝
- Insurance 도메인 작성
  - 보험 상품의 보장 항목별 응집도를 고려하여 클래스를 분리함.

## Uncompleted Tasks 😅

## To Reviewers 📢

jher235 committed:
[Feat] #14 Insurance 상품 상태 작성

jher235 committed:
[Feat] #14 Insurance 상품 큰병 진단비 작성

jher235 committed:
[Feat] #14 Insurance 상품 큰병 수술비 작성

jher235 committed:
[Feat] #14 Insurance 상품 상해 수술비 작성

jher235 committed:
[Feat] #14 Insurance 상품 질병 수술비 작성

jher235 committed:
[Feat] #14 Insurance 상품 일당 입원비 작성

jher235 committed:
[Feat] #14 Insurance 상품 사망 보험비 작성

jher235 committed:
[Feat] #14 Insurance 상품 상해 후유 장애 작성

jher235 committed:
[Feat] #14 Insurance 상품 기본 정보 작성
- 추후 보험사를 사용할 경우 보험사 테이블 분리 후 연관관계를 설정하도록 수정할 여지 존재

jher235 committed:
[Feat] #14 Insurance 상품 추가 정보 작성

jher235 committed:
[Feat] #14 Insurance 도메인 작성
- 응집도에 맞춰 핵심 정보, 큰병 진단비, 큰병 수술비, 질병 수술비, 상해 수술비, 장해, 사망 보험, 추가 정보를 각각 하위 클래스로 작성함.

jher235 committed:
[Chore] #14 오타 수정
- matueity_age -> maturity_age

jeong1112 commented on src/main/java/org/sopt/bofit/domain/insurance/entity/BasicInformation.java:
```diff
@@ -0,0 +1,35 @@
+@Embeddable
+public class BasicInformation {
+
+	@Column(name = "provider", length = 30)
+	private String provider;
```
보험사에 해당하는 필드인 것 같은데, 네이밍을 company와 같이 설정하는 것이 더 직관적일 것 같습니다!

jeong1112 commented on src/main/java/org/sopt/bofit/domain/insurance/entity/BasicInformation.java:
```diff
@@ -0,0 +1,35 @@
+	@Column(name = "product_type", length = 50)
+	private String productType;
+
+	@Column(name = "min_enrollment_age", nullable = false,
+		columnDefinition = "TINYINT UNSIGNED DEFAULT 0")
+	private Integer minEnrollmentAge;
+	@Column(name = "max_enrollment_age", nullable = false,
+		columnDefinition = "TINYINT UNSIGNED DEFAULT 0")
+	private Integer maxEnrollmentAge;
+
+	@Column(name = "premium", nullable = false,
+		columnDefinition = "MEDIUMINT UNSIGNED DEFAULT 0")
+	private Integer premium;
+
+	@Column(name = "maturity_age", nullable = false,
+		columnDefinition = "TINYINT UNSIGNED DEFAULT 0")
+	private Integer	maturityAge;
+
+	@Column(name = "payment_period_years", nullable = false,
+		columnDefinition = "TINYINT UNSIGNED DEFAULT 0")
+	private Integer paymentPeriodYears;
```
정수형 타입을 Wrapper 타입으로 받는 이점이 있나요?

jher235 replied:
사실 다른 것보다 추후 해당 값에 null 이 올 수 있다는 생각에 대부분의 경우 안전하게 Wrapper를 사용하는데요. 현재는 Insurance 엔티티에 정수 값 필드가 굉장히 많기 때문에 primitive type을 사용하는게 이점이 있을 것 같습니다. 이것도 수정하겠습니다~

jeong1112 commented on src/main/java/org/sopt/bofit/domain/insurance/entity/ExtraInformation.java:
```diff
@@ -0,0 +1,19 @@
+	@Column(name = "external_uri")
+	private String externalUri;
+
+	@Column(name = "informaion_source")
```
이 부분 information 오타인가요?

jeong1112 commented on src/main/java/org/sopt/bofit/domain/insurance/entity/MajorDiagnosisBenefit.java:
가독성을 위해 필드 선언 뒤에 줄바꿈 추가해주시면 좋을 것 같아요!

jeong1112 commented on src/main/java/org/sopt/bofit/domain/insurance/entity/MajorSurgeryBenefit.java:
이 부분도 마찬가지로 줄바꿈 있으면 좋겠네요

jeong1112 reviewed (approved):
고생하셨습니다! 아무래도 필드명 네이밍은 의학적인 부분이라 어려울 수 밖에 없는 것 같네요.. ㅎㅎ

jher235 committed:
[Chore] #14 가독성 향상을 위해 필드 공백 추가

jher235 committed:
[Chore] #14 보험 회사 필드명 수정
provider -> company

jher235 committed:
[Chore] #14 정수 값 필드를 Reference type에서 Primitive type으로 전환
- 바이트 수가 16 -> 4로 줄어들어 이득이 큼. 특히 현재 같이 정수 필드가 많은 경우.
- notnull을 DB에서 명시적으로 제시하고 있으므로 큰 부담이 없음.
